Userland/Libraries/LibJS/Interpreter.cpp

```diff
@@ -166,6 +166,9 @@ Value Interpreter::execute_statement(GlobalObject& global_object, const Statemen
     auto& block = static_cast<const ScopeNode&>(statement);
     enter_scope(block, scope_type, global_object);
 
+    if (scope_type == ScopeType::With)
+        vm().set_last_value({}, js_undefined());
```

Comment:
Let's not pollute `execute_statement()` with more ad hoc fixes - let's follow the spec closely instead:

https://tc39.es/ecma262/#sec-with-statement-runtime-semantics-evaluation

> Return Completion(UpdateEmpty(C, undefined)).

i.e. we just update `WithStatement::execute()` to `return interpreter.execute_statement(global_object, m_body).value_or(js_undefined());` - which is then picked up by this below:

```
if (!value.is_empty())
    vm().set_last_value({}, value);
```

Reply:
yeah unfortunately that wont work, as you can see set_last_value is *never* called if the value is empty, meaning we get the last value from before the with block, so we dont replace it with js_undefined (i have tried to follow the spec, but this is the cleanest solution i came up with, since we dont have real completions :sweat:)

Reply:
> as you can see set_last_value is never called if the value is empty

Yes, but WithStatement is supposed to never return an empty value for the non-exception case, that's the whole point...